scripts/train_dolphin_unsloth.py

```
@@ -575,13 +576,24 @@ def prepare_model_and_tokenizer(config: TrainingConfig) -> tuple[Any, Any]:
         load_kwargs["dtype"] = torch.float16
     else:
         load_kwargs["dtype"] = torch.float32
 
+    if torch.cuda.is_available():
+        try:
+            free_bytes, _total_bytes = torch.cuda.mem_get_info()
+            free_mb = int(free_bytes // (1024 * 1024))
+            budget_mb = max(1024, free_mb - 512)
+            load_kwargs["device_map"] = {"": 0}
+            load_kwargs["max_memory"] = {0: f"{budget_mb}MiB"}
+        except Exception:
```

Ruff (0.13.3)

Do not catch blind exception: `Exception`

(BLE001)